File: branches/Medusa/Source/Fantasy.Jobs/WCFSingletonService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;
using System.Threading;
using Fantasy.ServiceModel;

namespace Fantasy.Jobs
{
    public abstract class WCFSingletonService : AbstractService
    {
        private ServiceHost _serviceHost;

        public override void InitializeService()
        {
            
            Thread thread = ThreadFactory.CreateThread(() =>
            {
                _serviceHost = new ServiceHost(this);
                _serviceHost.Open();
            }).WithStart();
         
            thread.Join();
            base.InitializeService();
        }

        public override void UninitializeService()
        {

            base.UninitializeService();
            this._serviceHost.Close();
        }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/WCFSingletonService.java
package fantasy.jobs;

import fantasy.servicemodel.*;

public abstract class WCFSingletonService extends AbstractService
{
	private ServiceHost _serviceHost;

	@Override
	public void InitializeService()
	{

//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		Thread thread = ThreadFactory.CreateThread(() =>
		{
			_serviceHost = new ServiceHost(this);
			_serviceHost.Open();
		}
	   ).WithStart();

		thread.join();
		super.InitializeService();
	}

	@Override
	public void UninitializeService()
	{

		super.UninitializeService();
		this._serviceHost.Close();
	}
}